components/brave_vpn/resources/panel/components/toggle/index.tsx

```diff
@@ -1,22 +1,56 @@
 import * as React from 'react'
+import { LoaderIcon } from 'brave-ui/components/icons'
+
 import * as S from './style'
+import locale from '../../constants/locale'
+import { ConnectionState } from '../../api/panel_browser_api'
+import { useSelector, useDispatch } from '../../state/hooks'
+import * as Actions from '../../state/actions'
 
-interface Props {
-  isOn: boolean
-  onClick: () => void
+function useIsOnSelector () {
+  // We derive isOn from connectionStatus to be synchronized
+  const connectionStatus = useSelector(state => state.connectionStatus)
+  const isConnectionActive = [ConnectionState.CONNECTED, ConnectionState.CONNECTING].includes(connectionStatus)
+  return isConnectionActive
 }
 
-function Toggle (props: Props) {
+function Toggle () {
+  const dispatch = useDispatch()
+  const isOn = useIsOnSelector()
+  const status = useSelector(state => state.connectionStatus)
+
+  const onToggleClick = () => {
+    const activated = !isOn
+    if (activated) dispatch(Actions.connect())
+    else dispatch(Actions.disconnect())
+  }
+
   return (
-    <S.Box
-      type='button'
-      role='switch'
-      aria-checked={props.isOn}
-      onClick={props.onClick}
-      isActive={props.isOn}
-    >
-      <S.Circle isActive={props.isOn} />
-    </S.Box>
+    <>
+      <S.ToggleBox
+        type='button'
+        role='switch'
+        aria-checked={isOn}
+        onClick={onToggleClick}
+        isActive={isOn}
+      >
+        <S.Knob isActive={isOn} />
+      </S.ToggleBox>
+      <S.StatusBox>
+        {status === ConnectionState.CONNECTED && <S.ActiveIndicator />}
+        {status === ConnectionState.CONNECTING && <S.Loader><LoaderIcon /></S.Loader>}
+        {status === ConnectionState.DISCONNECTING && <S.Loader><LoaderIcon /></S.Loader>}
+        {status === ConnectionState.DISCONNECTED && <S.InActiveIndicator />}
+        {status === ConnectionState.CONNECT_FAILED && <S.FailedIndicator />}
+        <S.StatusText>
+          {status === ConnectionState.CONNECTED && locale.connectedLabel}
+          {status === ConnectionState.CONNECTING && `${locale.connectingLabel}...`}
+          {status === ConnectionState.DISCONNECTING && `${locale.disconnectingLabel}...`}
```

Comment:
Shouldn't the `...` should be part of the string - different languages might treat that differently. I believe we do that for other strings in the product. Also I believe it's better to use the ellipsis character `…`